### PR TITLE
Fetcher: Wait for bigger STH if started from the very end

### PR DESCRIPTION
**Pull request by pavelkalinnikov**
This change fixes a bug in Fetcher which causes Migrillian to finish work early instead of waiting continuously.

### CODE REVIEW

File: scanner/fetcher.go [hunk 1]
```diff
@@ -171,7 +171,17 @@ func (f *Fetcher) genRanges(ctx context.Context) <-chan fetchRange {
 		defer close(ranges)
 		start, end := f.opts.StartIndex, f.opts.EndIndex
 
-		for start < end {
+		for start < end || f.opts.Continuous {
+			// In continuous mode wait for bigger STH every time we reach the end,
+			// including, possibly, the very first iteration.
+			if start == end { // Implies f.opts.Continuous == true.
+				if err := f.updateSTH(ctx); err != nil {
+					glog.Warningf("STH update cancelled: %v", err)
```
Comment (Martin2112) [inline scanner/fetcher.go]: Message could be clearer if there are other errors that can happen than the context being cancelled. Could mention it was waiting for log to increase in size.

Comment (Martin2112) [inline scanner/fetcher.go]: Message could be clearer if there are other errors that can happen than the context being cancelled. Could mention it was waiting for log to increase in size.

Comment (pavelkalinnikov) [reply scanner/fetcher.go]: Done.

**Comment by Martin2112** [comment]
This PR isn't showing up for me atm.

**Comment by pavelkalinnikov** [comment]
Do you mean you can't pull it? Worked for me earlier today.

**Comment by Martin2112** [comment]
The first time I tried to review I couldn't see it on the list and I got a
404 going to it directly. It seemed to work later..

Martin

**Pull request merged**